app/api/api_v1/routers/admin.py

```diff
@@ -97,11 +97,23 @@ async def update_document(
         }
 
         for language in meta_data.languages:
-            lang = (
-                db.query(Language)
-                .filter(Language.language_code == language)
-                .one_or_none()
-            )
+            if len(language) == 2:
+                # The part1_code is the iso639-1 two letter language code.
+                # This is the way languages are detected in the pipeline.
+                lang = (
+                    db.query(Language)
+                    .filter(Language.part1_code == language)
+                    .one_or_none()
+                )
+            elif len(language) == 3:
+                lang = (
+                    db.query(Language)
+                    .filter(Language.language_code == language)
+                    .one_or_none()
+                )
```

Comment:
I thought we were falling back from language_code to part2_code?

Reply:
Logging a warning if the language is None. This is the only scenario we need to be aware of. If we retrieve a language and it doesn't exist then we will add it.